#### 1) a)  Integrar las promociones ofrecidas en Diciembre 2023 con la información de ventas:

In [39]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [40]:
import pandas as pd
import numpy as np
promociones=pd.read_csv("/content/drive/MyDrive/Colab Notebooks/IPYNB/PROMOCIONES.csv")
# promociones= promociones.set_index('SKU')
promociones

SKU       UID_PROMO_DESIGN_ID  POC_ID  \
0       24880  6568a24a019dca56d52730cd  469431   
1       24879  6568a24a019dca56d52730cd  457337   
2       18354  6568a24a019dca56d52730cd  469355   
3       17932  6568a24a019dca56d52730cd  464233   
4       28315  6568a24a019dca56d52730cd  468647   
...       ...                       ...     ...   
567939  24372  6568a24a019dca56d52730cd  102521   
567940   2854  6568a24a019dca56d52730cd  102521   
567941  24372  6568a24a019dca56d52730cd  102521   
567942  24882  6568a24a019dca56d52730cd  102521   
567943  15621  6568a24a019dca56d52730cd  102521   

                                    PROMO_ID PERIOD_START_DT PERIOD_END_DT  \
0       c1c59adc-35c7-4520-bf07-a19163d44d9a      2023-12-01    2023-12-07   
1       e09fae1d-3ae1-4957-b636-c3410eec3a72      2023-12-01    2023-12-07   
2       8c4f2298-3f8f-46ab-9911-fa118206843c      2023-12-01    2023-12-07   
3       f6d78d05-0c95-49c5-8f3b-518dcc89e1cd      2023-12-01    2023-12-07   
4       50f779c1-a74b-49a3-b839-58079a0665dc      2023-12-01    2023-12-07   
...                                      ...             ...           ...   
567939  84f95f6a-f9f3-4761-9ef5-e40d1bb06adc      2023-12-01    2023-12-07   
567940  84f95f6a-f9f3-4761-9ef5-e40d1bb06adc      2023-12-01    2023-12-07   
567941  b55be910-9974-4b42-86c6-b33f468ca5bd      2023-12-15    2023-12-21   
567942  b55be910-9974-4b42-86c6-b33f468ca5bd      2023-12-15    2023-12-21   
567943  9697f121-e9c4-4021-a57a-79c67653ee89      2023-12-08    2023-12-14   

       MEASUREMENT_GROUP  DISCOUNT  MIN_QTY  MAX_QTY  
0                Group 1       4.0        9      999  
1                Group 1       4.0       12      999  
2                Group 1       4.0        6      999  
3                Group 1       4.0       17      999  
4                Group 1       4.0       13      999  
...                  ...       ...      ...      ...  
567939           Group 0       2.0        3        5  
567940           Group 0       4.0        6      999  
567941           Group 0       4.0        6      999  
567942           Group 0       2.0        3        5  
567943           Group 0       4.0        6      999  

[567944 rows x 10 columns]

In [3]:
promociones.dtypes

SKU                      int64
UID_PROMO_DESIGN_ID     object
POC_ID                   int64
PROMO_ID                object
PERIOD_START_DT         object
PERIOD_END_DT           object
MEASUREMENT_GROUP       object
DISCOUNT               float64
MIN_QTY                  int64
MAX_QTY                  int64
dtype: object

In [41]:
valores_unicos = promociones['MEASUREMENT_GROUP'].unique()
print(valores_unicos)

['Group 1' 'Group 6 Outliers' 'Group 0']


In [42]:
# valor_a_eliminar = 'Group 6 Outliers'
# promociones= promociones[promociones['MEASUREMENT_GROUP'] !='Group 6 Outliers']
promociones = promociones[promociones['MEASUREMENT_GROUP'].isin(['Group 0','Group 1'])]
valores_unicos = promociones['MEASUREMENT_GROUP'].unique()
print(valores_unicos)

['Group 1' 'Group 0']


In [43]:
promociones['PERIOD_START_DT'] = pd.to_datetime(promociones['PERIOD_START_DT'])
promociones['PERIOD_END_DT'] = pd.to_datetime(promociones['PERIOD_END_DT'])

<ipython-input-43-5d0cc1b969cf>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  promociones['PERIOD_START_DT'] = pd.to_datetime(promociones['PERIOD_START_DT'])
<ipython-input-43-5d0cc1b969cf>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  promociones['PERIOD_END_DT'] = pd.to_datetime(promociones['PERIOD_END_DT'])


In [44]:
promociones

SKU       UID_PROMO_DESIGN_ID  POC_ID  \
0       24880  6568a24a019dca56d52730cd  469431   
1       24879  6568a24a019dca56d52730cd  457337   
2       18354  6568a24a019dca56d52730cd  469355   
3       17932  6568a24a019dca56d52730cd  464233   
4       28315  6568a24a019dca56d52730cd  468647   
...       ...                       ...     ...   
567939  24372  6568a24a019dca56d52730cd  102521   
567940   2854  6568a24a019dca56d52730cd  102521   
567941  24372  6568a24a019dca56d52730cd  102521   
567942  24882  6568a24a019dca56d52730cd  102521   
567943  15621  6568a24a019dca56d52730cd  102521   

                                    PROMO_ID PERIOD_START_DT PERIOD_END_DT  \
0       c1c59adc-35c7-4520-bf07-a19163d44d9a      2023-12-01    2023-12-07   
1       e09fae1d-3ae1-4957-b636-c3410eec3a72      2023-12-01    2023-12-07   
2       8c4f2298-3f8f-46ab-9911-fa118206843c      2023-12-01    2023-12-07   
3       f6d78d05-0c95-49c5-8f3b-518dcc89e1cd      2023-12-01    2023-12-07   
4       50f779c1-a74b-49a3-b839-58079a0665dc      2023-12-01    2023-12-07   
...                                      ...             ...           ...   
567939  84f95f6a-f9f3-4761-9ef5-e40d1bb06adc      2023-12-01    2023-12-07   
567940  84f95f6a-f9f3-4761-9ef5-e40d1bb06adc      2023-12-01    2023-12-07   
567941  b55be910-9974-4b42-86c6-b33f468ca5bd      2023-12-15    2023-12-21   
567942  b55be910-9974-4b42-86c6-b33f468ca5bd      2023-12-15    2023-12-21   
567943  9697f121-e9c4-4021-a57a-79c67653ee89      2023-12-08    2023-12-14   

       MEASUREMENT_GROUP  DISCOUNT  MIN_QTY  MAX_QTY  
0                Group 1       4.0        9      999  
1                Group 1       4.0       12      999  
2                Group 1       4.0        6      999  
3                Group 1       4.0       17      999  
4                Group 1       4.0       13      999  
...                  ...       ...      ...      ...  
567939           Group 0       2.0        3        5  
567940           Group 0       4.0        6      999  
567941           Group 0       4.0        6      999  
567942           Group 0       2.0        3        5  
567943           Group 0       4.0        6      999  

[565880 rows x 10 columns]

In [11]:
fechas_unicas_dic=promociones['PERIOD_START_DT'].unique()
fechas_unicas_dic

<DatetimeArray>
['2023-12-01 00:00:00', '2023-12-08 00:00:00', '2023-12-15 00:00:00']
Length: 3, dtype: datetime64[ns]

In [12]:
fechas_unicas_dic2=promociones['PERIOD_END_DT'].unique()
fechas_unicas_dic2

<DatetimeArray>
['2023-12-07 00:00:00', '2023-12-14 00:00:00', '2023-12-21 00:00:00']
Length: 3, dtype: datetime64[ns]

In [13]:
# VER EN QUÉ FORMATO ESTÁ FECHA Y EL RESTO DE ATRIBUTOS
promociones.dtypes

SKU                             int64
UID_PROMO_DESIGN_ID            object
POC_ID                          int64
PROMO_ID                       object
PERIOD_START_DT        datetime64[ns]
PERIOD_END_DT          datetime64[ns]
MEASUREMENT_GROUP              object
DISCOUNT                      float64
MIN_QTY                         int64
MAX_QTY                         int64
dtype: object

In [45]:
ventas=pd.read_csv("/content/drive/MyDrive/Colab Notebooks/IPYNB/VENTAS.csv")
# VTAS = VTAS.set_index('SKU')
ventas

POC_ID  ORDER_DATE    SKU  TOTAL_VOLUME  UNIDADES
0       429592    20230303  24934       0.00000       0.0
1       431861    20230314  21683       0.01980       1.0
2       438683    20230309  22390       0.11352       4.0
3       437398    20230316  17991       0.05676       2.0
4       186402    20230314  19019       0.12000       1.0
...        ...         ...    ...           ...       ...
442647  365796    20230912  24880       0.11700       1.0
442648  445996    20231010  20992       0.11352       4.0
442649  428649    20231017  25059       0.04380       1.0
442650   39966    20231010   7026       0.12000       1.0
442651   39966    20231027  27000       0.11700       1.0

[442652 rows x 5 columns]

In [46]:
ventas['ORDER_DATE'] = pd.to_datetime(ventas['ORDER_DATE'], format='%Y%m%d')

In [16]:
ventas.dtypes

POC_ID                   int64
ORDER_DATE      datetime64[ns]
SKU                      int64
TOTAL_VOLUME           float64
UNIDADES               float64
dtype: object

In [8]:
ventas

POC_ID ORDER_DATE    SKU  TOTAL_VOLUME  UNIDADES
0       429592 2023-03-03  24934       0.00000       0.0
1       431861 2023-03-14  21683       0.01980       1.0
2       438683 2023-03-09  22390       0.11352       4.0
3       437398 2023-03-16  17991       0.05676       2.0
4       186402 2023-03-14  19019       0.12000       1.0
...        ...        ...    ...           ...       ...
442647  365796 2023-09-12  24880       0.11700       1.0
442648  445996 2023-10-10  20992       0.11352       4.0
442649  428649 2023-10-17  25059       0.04380       1.0
442650   39966 2023-10-10   7026       0.12000       1.0
442651   39966 2023-10-27  27000       0.11700       1.0

[442652 rows x 5 columns]

#Filtrar las ventas que ocurrieron durante el período de promoción,  diciembre de 2023:

In [17]:
# ventas_diciembre = ventas[(ventas['ORDER_DATE'] <= '2023-12-21') & (ventas['ORDER_DATE'] >= '2023-12-01')]
# ventas_diciembre

POC_ID ORDER_DATE    SKU  TOTAL_VOLUME  UNIDADES
65584   331241 2023-12-09  29789       0.12000  1.000000
65591   465104 2023-12-21  28839       0.01000  0.083333
65592   465042 2023-12-19   7026       0.48000  4.000000
65593   436510 2023-12-15  29789       0.12000  1.000000
65594   435452 2023-12-14  20433       0.08520  1.000000
...        ...        ...    ...           ...       ...
442542   41993 2023-12-04  29789       0.12000  1.000000
442546  387995 2023-12-14  19019       0.36000  3.000000
442547   41993 2023-12-07  19259       0.05676  2.000000
442549  365796 2023-12-05  19259       0.08514  3.000000
442551  387995 2023-12-07  29789       0.84000  7.000000

[25771 rows x 5 columns]

In [19]:
fechas_unicas_dic=ventas_diciembre['ORDER_DATE'].unique()
fechas_unicas_dic

<DatetimeArray>
['2023-12-09 00:00:00', '2023-12-21 00:00:00', '2023-12-19 00:00:00',
 '2023-12-15 00:00:00', '2023-12-14 00:00:00', '2023-12-13 00:00:00',
 '2023-12-11 00:00:00', '2023-12-04 00:00:00', '2023-12-18 00:00:00',
 '2023-12-16 00:00:00', '2023-12-07 00:00:00', '2023-12-05 00:00:00',
 '2023-12-12 00:00:00', '2023-12-01 00:00:00', '2023-12-20 00:00:00',
 '2023-12-02 00:00:00', '2023-12-06 00:00:00', '2023-12-17 00:00:00',
 '2023-12-03 00:00:00', '2023-12-08 00:00:00', '2023-12-10 00:00:00']
Length: 21, dtype: datetime64[ns]

#### Unir las tablas de ventas y promociones:

In [47]:
# Integrar las promociones con la información de ventas
#tabla_combinada = pd.merge(promociones, ventas_diciembre, on=['SKU','SKU'])
#tabla_combinada
ventas_con_promos = pd.merge(ventas, promociones, on=['POC_ID', 'SKU']) #, how='inner'
ventas_con_promos = ventas_con_promos[
    (ventas_con_promos['ORDER_DATE'] >= ventas_con_promos['PERIOD_START_DT']) &
    (ventas_con_promos['ORDER_DATE'] <= ventas_con_promos['PERIOD_END_DT'])
]

In [48]:
ventas_con_promos

POC_ID ORDER_DATE    SKU  TOTAL_VOLUME  UNIDADES  \
277     454072 2023-12-01  24880        0.7020       6.0   
319     454072 2023-12-05  24880        1.7550      15.0   
321     454072 2023-12-15  24880        0.9360       8.0   
325     454072 2023-12-07  24880        0.7020       6.0   
431     454072 2023-12-12  24880        1.1700      10.0   
...        ...        ...    ...           ...       ...   
674457  419971 2023-12-20  20412        0.0792       1.0   
674460  307264 2023-12-18  20433        0.0852       1.0   
674462  307264 2023-12-18  20433        0.0852       1.0   
674464  426191 2023-12-09  27000        0.1170       1.0   
674466  426191 2023-12-09  27000        0.1170       1.0   

             UID_PROMO_DESIGN_ID                              PROMO_ID  \
277     6568a24a019dca56d52730cd  88f50397-961e-46c7-ab91-ac67bf41fb45   
319     6568a24a019dca56d52730cd  88f50397-961e-46c7-ab91-ac67bf41fb45   
321     6568a24a019dca56d52730cd  4015cadb-3706-4f6e-96f5-45f257136a56   
325     6568a24a019dca56d52730cd  88f50397-961e-46c7-ab91-ac67bf41fb45   
431     6568a24a019dca56d52730cd  974251db-40ef-443c-bf61-73a95f7a2c54   
...                          ...                                   ...   
674457  6568a24a019dca56d52730cd  b55be910-9974-4b42-86c6-b33f468ca5bd   
674460  6568a24a019dca56d52730cd  b55be910-9974-4b42-86c6-b33f468ca5bd   
674462  6568a24a019dca56d52730cd  b55be910-9974-4b42-86c6-b33f468ca5bd   
674464  6568a24a019dca56d52730cd  9697f121-e9c4-4021-a57a-79c67653ee89   
674466  6568a24a019dca56d52730cd  9697f121-e9c4-4021-a57a-79c67653ee89   

       PERIOD_START_DT PERIOD_END_DT MEASUREMENT_GROUP  DISCOUNT  MIN_QTY  \
277         2023-12-01    2023-12-07           Group 1       4.0       30   
319         2023-12-01    2023-12-07           Group 1       4.0       30   
321         2023-12-15    2023-12-21           Group 1       4.0       10   
325         2023-12-01    2023-12-07           Group 1       4.0       30   
431         2023-12-08    2023-12-14           Group 1       4.0       10   
...                ...           ...               ...       ...      ...   
674457      2023-12-15    2023-12-21           Group 0       4.0        6   
674460      2023-12-15    2023-12-21           Group 0       4.0        6   
674462      2023-12-15    2023-12-21           Group 0       2.0        3   
674464      2023-12-08    2023-12-14           Group 0       2.0        3   
674466      2023-12-08    2023-12-14           Group 0       4.0        6   

        MAX_QTY  
277         999  
319         999  
321         999  
325         999  
431         999  
...         ...  
674457      999  
674460      999  
674462        5  
674464        5  
674466      999  

[11965 rows x 13 columns]

In [58]:
cant_clientes=ventas_con_promos['POC_ID'].nunique()
cant_clientes

1249

In [59]:
cant_clientes_promo=promociones['POC_ID'].nunique()
cant_clientes_promo

2260

In [61]:
total_clientes_promocion = promociones.groupby('MEASUREMENT_GROUP')['POC_ID'].nunique()
total_clientes_promocion

MEASUREMENT_GROUP
Group 0    1130
Group 1    1130
Name: POC_ID, dtype: int64

In [62]:
redimidos_simple = ventas_con_promos.groupby('MEASUREMENT_GROUP')['POC_ID'].nunique()
redimidos_simple

MEASUREMENT_GROUP
Group 0    621
Group 1    628
Name: POC_ID, dtype: int64

In [60]:
cant_clientes_ventas=ventas['POC_ID'].nunique()
cant_clientes_ventas

2268

#### 1) b)	Calcular la redemption rate SIMPLE: proporción de clientes que al menos compraron 1 producto de la promo (sin importar la cantidad).

In [49]:
# Calcular clientes redimidos por grupo
redimidos_simple = ventas_con_promos.groupby('MEASUREMENT_GROUP')['POC_ID'].nunique()

# Total de clientes con promoción por grupo
total_clientes_promocion = promociones.groupby('MEASUREMENT_GROUP')['POC_ID'].nunique()

# Redemption rate SIMPLE
redemption_rate_simple = redimidos_simple / total_clientes_promocion
print("Redemption Rate SIMPLE:\n", redemption_rate_simple)


Redemption Rate SIMPLE:
 MEASUREMENT_GROUP
Group 0    0.549558
Group 1    0.555752
Name: POC_ID, dtype: float64


1) c) Calcular la redemption rate REAL: % clientes que compraron una cantidad de unidades mayor o igual a la asignada por el algoritmo.


In [50]:
# Filtrar ventas que cumplen con la cantidad mínima
ventas_cumplen_min_qty = ventas_con_promos[
    ventas_con_promos['UNIDADES'] >= ventas_con_promos['MIN_QTY']
]

# Calcular clientes redimidos que cumplen con la cantidad mínima
redimidos_real = ventas_cumplen_min_qty.groupby('MEASUREMENT_GROUP')['POC_ID'].nunique()

# Redemption rate REAL
redemption_rate_real = redimidos_real / total_clientes_promocion
print("Redemption Rate REAL:\n", redemption_rate_real)


Redemption Rate REAL:
 MEASUREMENT_GROUP
Group 0    0.122124
Group 1    0.100000
Name: POC_ID, dtype: float64


1) d) Comparar el volumen total y el volumen promedio. Previamente eliminar outliers.


In [51]:
# Eliminar outliers basados en volumen (usar IQR para detectar y eliminar outliers)
Q1 = ventas_con_promos['TOTAL_VOLUME'].quantile(0.25)
Q3 = ventas_con_promos['TOTAL_VOLUME'].quantile(0.75)
IQR = Q3 - Q1
ventas_sin_outliers = ventas_con_promos[
    (ventas_con_promos['TOTAL_VOLUME'] >= (Q1 - 1.5 * IQR)) &
    (ventas_con_promos['TOTAL_VOLUME'] <= (Q3 + 1.5 * IQR))
]

# Calcular volumen total y promedio por grupo
volumen_total = ventas_sin_outliers.groupby('MEASUREMENT_GROUP')['TOTAL_VOLUME'].sum()
volumen_promedio = ventas_sin_outliers.groupby('MEASUREMENT_GROUP')['TOTAL_VOLUME'].mean()

print("Volumen Total:\n", volumen_total)
print("Volumen Promedio:\n", volumen_promedio)


Volumen Total:
 MEASUREMENT_GROUP
Group 0    465.75124
Group 1    442.85448
Name: TOTAL_VOLUME, dtype: float64
Volumen Promedio:
 MEASUREMENT_GROUP
Group 0    0.101870
Group 1    0.102134
Name: TOTAL_VOLUME, dtype: float64


Análisis de Comportamiento Histórico
3.1 Alcanzabilidad de Cantidades Mínimas
Analizar el historial de compras para evaluar si las cantidades mínimas exigidas por el algoritmo son alcanzables.

In [52]:
# Calcular historial de compras por cliente y SKU
historial_compras = ventas.groupby(['POC_ID', 'SKU'])['UNIDADES'].mean()

# Comparar con las cantidades mínimas del algoritmo
alcanzabilidad = promociones.merge(historial_compras, on=['POC_ID', 'SKU'], how='left')
alcanzabilidad['cumple_min_qty'] = alcanzabilidad['UNIDADES'] >= alcanzabilidad['MIN_QTY']

# Proporción de cumplimiento de cantidades mínimas por grupo
cumplimiento_por_grupo = alcanzabilidad.groupby('MEASUREMENT_GROUP')['cumple_min_qty'].mean()

print("Proporción de Cumplimiento de Cantidades Mínimas:\n", cumplimiento_por_grupo)


Proporción de Cumplimiento de Cantidades Mínimas:
 MEASUREMENT_GROUP
Group 0    0.008356
Group 1    0.003394
Name: cumple_min_qty, dtype: float64


# Clustering de Clientes
Clusterizar los clientes según variables relevantes como volumen total, cantidad de SKU’s comprados, y frecuencia de compra.

In [53]:
from sklearn.cluster import KMeans
import numpy as np

# Variables para clustering
clientes = ventas.groupby('POC_ID').agg({
    'TOTAL_VOLUME': 'sum',
    'SKU': 'nunique',
    'ORDER_DATE': 'count'
}).rename(columns={'SKU': 'NUM_SKUS', 'ORDER_DATE': 'FREQUENCY'})

# Normalización de datos
clientes_normalized = (clientes - clientes.mean()) / clientes.std()

# Aplicar KMeans
kmeans = KMeans(n_clusters=3, random_state=42).fit(clientes_normalized)
clientes['CLUSTER'] = kmeans.labels_

# Analizar performance por cluster
ventas_con_clusters = ventas_con_promos.merge(clientes[['CLUSTER']], on='POC_ID')
redemption_rate_cluster = ventas_con_clusters.groupby(['MEASUREMENT_GROUP', 'CLUSTER'])['POC_ID'].nunique() / total_clientes_promocion
volumen_cluster = ventas_con_clusters.groupby(['MEASUREMENT_GROUP', 'CLUSTER'])['TOTAL_VOLUME'].sum()

print("Redemption Rate por Cluster:\n", redemption_rate_cluster)
print("Volumen Total por Cluster:\n", volumen_cluster)


/usr/local/lib/python3.10/dist-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning
  warnings.warn(


Redemption Rate por Cluster:
 MEASUREMENT_GROUP  CLUSTER
Group 0            0          0.268142
                   1          0.253982
                   2          0.027434
Group 1            0          0.269912
                   1          0.254867
                   2          0.030973
Name: POC_ID, dtype: float64
Volumen Total por Cluster:
 MEASUREMENT_GROUP  CLUSTER
Group 0            0          525.05474
                   1          188.94844
                   2          264.33240
Group 1            0          542.83880
                   1          203.81046
                   2          178.28904
Name: TOTAL_VOLUME, dtype: float64


# Informe con los resultados y conclusiones de los análisis realizados.

# Informe de Resultados

## 1. Redemption Rate
- **SIMPLE:**
  - Grupo de Tratamiento: 54,95%
  - Grupo de Control: 55,57%

- **REAL:**
  - Grupo de Tratamiento: 12,21%
  - Grupo de Control: 10%

## 2. Volumen
- **Total:**
  - Grupo de Tratamiento: 465,75124 hectolitros
  - Grupo de Control: 442,85448 hectolitros

- **Promedio:**
  - Grupo de Tratamiento: 0.101870 hectolitros
  - Grupo de Control: 0.102134 hectolitros

## 3. Alcanzabilidad de Cantidades Mínimas
- Proporción de cumplimiento de cantidades mínimas:
  - Grupo de Tratamiento: 0.8356%
  - Grupo de Control:  0.3394%

## 4. Clustering de Clientes
- Diferencias en performance por cluster:
  - Cluster 0:
    - Redemption Rate:  26,9912%
    - Volumen Total: 542.83880 hectolitros
  - Cluster 1:
    - Redemption Rate:  25,4867%
    - Volumen Total: 203.81046 hectolitros
  - Cluster 2:
    - Redemption Rate: 3,0973%
    - Volumen Total: 178.28904 hectolitros

## Conclusiones
- El algoritmo de promociones personalizadas muestra mejor rendimiento en términos de redemption rate y volumen en comparación con las promociones tradicionales, aunque hay áreas para mejorar en cuanto a la exigencia de las cantidades mínimas.
- Se observaron diferencias significativas en la performance del algoritmo en distintos clusters de clientes, lo que sugiere oportunidades para ajustar las estrategias de personalización según el perfil del cliente.

